ui/client/common/ProcessUtils.js

```diff
@@ -50,14 +50,17 @@ class ProcessUtils {
   //see BranchEndDefinition.artificialNodeId
   findVariablesForBranches = (nodeResults) => (nodeId) => {
     //we find all nodes matching pattern encoding branch and edge and extract branch id
+    const escapedNodeId = this.escapeId(nodeId)
     return _.transform(nodeResults || {}, function(result, value, key) {
-      const branch = key.match(new RegExp(`^\\$edge-(.*)-${nodeId}$`))
+      const branch = key.match(new RegExp(`^\\$edge-(.*)-${escapedNodeId}$`))
       if (branch && branch.length > 1) {
         result[branch[1]] = value.variableTypes
       }
     }, {})
   }
 
+  escapeId = (id) => id.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
```

Comment:
escapeForRegexp? Or sth like that - because it will be different escaping for e.g. URL? And let's add link to source of this expression :)